app/uk/gov/hmrc/tradergoodsprofilesdatastore/models/RecordsSummary.scala

```diff
@@ -16,27 +16,20 @@
 
 package uk.gov.hmrc.tradergoodsprofilesdatastore.models
 
-import play.api.libs.json.{Format, JsSuccess, JsValue, Json, Reads, Writes}
+import play.api.libs.json.{Json, OFormat}
 
-case class CheckRecords(
-  eori: String
-)
-
-object CheckRecords {
+import java.time.Instant
 
-  implicit val writes: Writes[CheckRecords] = (checkRecords: CheckRecords) =>
-    Json.obj(
-      "eori" -> checkRecords.eori
-    )
+case class RecordsSummary(
```

Comment:
Given that `recordsStored` and `recordsToStore` are only really valid when `recordsUpdating` is `true`, I think it'd be best to model this a little differently. Something like:

```scala
final case class RecordsSummary(
  eori: String,
  currentUpdate: Option[RecordsSummary.Update],
  lastUpdated: Instant
) {
  def isUpdating: Boolean = currentUpdate.isDefined
}

object RecordSummary {
  
  final case class Update(recordsStored: Int, recordsToStore: Int)

  object Update {
    implicit lazy val format: OFormat[Update] = Json.format
  }

  implicit lazy val format: OFormat[RecordsSummary] = Json.format
}
```

Reply:
i agree